src/sentry/incidents/logic.py

```diff
@@ -1505,7 +1505,7 @@ def schedule_update_project_config(alert_rule: AlertRule, projects: Sequence[Pro
     ):
         return
 
-    if alert_rule.is_custom_metric:
+    if OndemandMetricSpec.check(alert_rule.snuba_query.aggregate, alert_rule.snuba_query.query):
```

Comment:
I think we might need an additional utility function on `snuba_query` that also checks `snuba_query.dataset`. That should probably not reside on the `SnubaQuery` model, but rather somewhere in the `metrics_extraction` module.

We run that same check in `convert_alert_to_metric`, but it's too easy to miss in other cases where we may need it.

This should resolve the TODO inside `check`.

Reply:
Added a util
https://github.com/getsentry/sentry/blob/bcc644e7084dcee4c6dd290a551b8db9bdba91f4/src/sentry/snuba/metrics/metric_extraction.py#L241-L245